static_frame/core/yarn.py

```diff
@@ -38,21 +38,27 @@
 from static_frame.core.style_config import StyleConfig
 from static_frame.core.util import DTYPE_OBJECT
 from static_frame.core.util import NAME_DEFAULT
-from static_frame.core.util import IndexConstructor
-from static_frame.core.util import IndexConstructors
-from static_frame.core.util import IndexInitializer
-from static_frame.core.util import NameType
 from static_frame.core.util import TILocSelector
+from static_frame.core.util import TIndexCtorSpecifier
+from static_frame.core.util import TIndexCtorSpecifiers
+from static_frame.core.util import TIndexInitializer
 from static_frame.core.util import TLabel
 from static_frame.core.util import TLocSelector
+from static_frame.core.util import TName
 from static_frame.core.util import is_callable_or_mapping
 
 if tp.TYPE_CHECKING:
-    NDArrayAny = np.ndarray[tp.Any, tp.Any] # pylint: disable=W0611 #pragma: no cover
-    DtypeAny = np.dtype[tp.Any] # pylint: disable=W0611 #pragma: no cover
+    TNDArrayAny = np.ndarray[tp.Any, tp.Any] # pylint: disable=W0611 #pragma: no cover
+    TDtypeAny = np.dtype[tp.Any] # pylint: disable=W0611 #pragma: no cover
 
+TSeriesAny = Series[tp.Any, tp.Any]
+TFrameAny = Frame[tp.Any, tp.Any, tp.Unpack[tp.Tuple[tp.Any, ...]]] # type: ignore[type-arg]
+TBusAny = Bus[tp.Any]
 
-class Yarn(ContainerBase, StoreClientMixin):
+#-------------------------------------------------------------------------------
+TVIndex = tp.TypeVar('TVIndex', bound=IndexBase, default=tp.Any)
```

Comment:
Shouldn't bound be `np.object_`? You can't have a yarn of anything other than `sf.Frame` or `sf.Series`

Reply:
No, I don't think so. This typevar is for the `Index` of the `Yarn`, which can be any Index. Separately, thanks to your feedback, we are now using `_series: TSeriesObject`, which is what I think you were thinking of here.